offer-manager-domain/src/main/groovy/life/qbic/business/products/copy/CopyProduct.groovy

```diff
@@ -12,9 +15,68 @@ import life.qbic.datamodel.dtos.business.ProductId
  * @since: 1.0.0
  *
  */
-class CopyProduct implements CopyProductInput {
+class CopyProduct implements CopyProductInput, CreateProductOutput {
+
+    private final CopyProductDataSource dataSource
+    private final CopyProductOutput output
+    private final CreateProductInput createProductInput
+
+    /**
+     * The only constructor for this use case
+     * @param dataSource - a data source that provides mandatory functionality
+     * @param output - an output that provides mandatory functionality
+     * @param createProductInput - a CreateProduct use case that is used to create the product
+     */
+    CopyProduct(CopyProductDataSource dataSource, CopyProductOutput output, CreateProductInput createProductInput) {
+        this.dataSource = dataSource
+        this.output = output
+        this.createProductInput = createProductInput
+    }
+
+    /**
+     * creates a copy of an existing product
+     * @param product The modified product information
+     * @since 1.0.0
+     */
     @Override
-    void copy(ProductId productId) {
+    void copyModified(Product product) {
+        throw new RuntimeException("Not implemented")
+        //TODO
+        //1. retrieve product from db
+        //2. construct new product with missing information filled from the db
+        //3. change product identifier
+        //4. call the CreateProduct use case
```

Comment:
you can call the `CreateProduct` use case simply by the `CopyProductOutput` interface of the `CopyProduct` use case. You implement the `CopyProductOutput` in the `CreateProduct` use case it solves how you can call the use case without a circular dependency